src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -272,10 +276,49 @@ impl BDA {
     where
         F: Read + Seek + SyncAll,
     {
-        StaticHeader::setup(f).map(|sh| sh.map(|sh| (sh.pool_uuid, sh.dev_uuid)))
+        match StaticHeader::setup(f) {
+            Ok(Some(SetupResult::Ok(sh))) => Ok(Some((sh.pool_uuid, sh.dev_uuid))),
+            Ok(Some(SetupResult::OkWithError(sh, err))) => {
+                setup_warn(&sh, err);
+                Ok(Some((sh.pool_uuid, sh.dev_uuid)))
+            }
+            Ok(None) => Ok(None),
+            Err(err) => Err(err),
+        }
     }
 }
 
+fn setup_warn(header: &StaticHeader, err: StratisError) {
+    warn!(
+        "Experienced an I/O error while attempting to repair an ill-formed, \
+         unreadable, or stale signature block: {:?}.\n\
+         Returning device \"{:?}\" from pool \"{:?}\".",
```

Comment:
How about ```Read and returned static header {:?}, sh``` for the second sentence. We implemented ```Debug``` for ```StaticHeader``` so that is doable; we had to implement it explicitly since the Debug info for UUIDs is far too long.

Reply:
I updated the text but not sure if it was how you had it in mind?